### ISSUE TITLE
Bug: Updating Profile Error

### ISSUE THREAD

**Issue by leonkoech** [opportunity-hack/evs#74]
## Description

Editing my DOB and height in the profile section results in an "Unexpected Server Error" API response, preventing the data from being saved.

## Steps to Reproduce

1. Access the profile editing page.
2. the DOB and height fields.
3. Submit the form.
4. Observe the "Unexpected Server Error" message.
5. Verify that the DOB and height remain unchanged.

### Request Payload:
```
username=leonkoech
name=Leon+Kipkoech
phone=786667486733333336674
mailingList=on
birthdate=1998-07-02
height=5.7
yearsOfExperience=0
currentPassword=
newPassword=
```
### Response:
`{"message":"Unexpected Server Error"}`

## Additional Information:

Browser: Chrome
Operating System: Windows 11

## Logs
You can find the logs [here ](https://drive.google.com/file/d/1LZSumgkADcTuBnbMfEbgBUVvD-FUdCEK/view?usp=sharing)

**Comment by parkerdavis1** [comment]
Another good catch and easy fix. The DB schema accepts integers and the zod validation was just checking that it was a number. Fix incoming.

Question: Does anyone have a more clear error message than “Height must be an integer in inches”? @gregv @benhsm 

As I think more about this field, asking users to calculate their heights in inches may be an error prone ask. An enhancement could be to split this to be a ‘feet’ field and an ‘inches’ field . This could be stored in the database with the number inputs from the fields being saved literally, or the inches calculation could be done before storing, and then converted back when displaying the information in the UI. Any thoughts?

**Comment by gregv** [comment]
Storing in inches but asking ft+in is the most straightforward way of asking for locales that don't use the metric system - I'm down to switch to this and then store/transpose this in inches to make it easy @parkerdavis1

**Comment by parkerdavis1** [comment]
Wow, I think I just experienced the joys of using pre 1.0 software (conform-to). There was a bug with how the app was treating numbers and number inputs. Before it would accept non-numerical strings as input to for number inputs, parse and submit them to zod as empty strings, then zod would turn the empty strings into 0, not throwing an error. 

I could not replicate the problems in a fresh install of the epic stack (which has been updated a lot since this project was started) so this clued me into the fact that it might be a bug. The only way I found to fix it was to update zod (3.21.4 --> 3.22.4) and conform-to/zod (0.7.0 -->  0.9.1) and change the number inputs to text inputs. 

Now :
1. when you input non-number text, zod can “see” the values properly and throw appropriate errors. ✅
2. when you input numbers, including 0, it saves them as numbers, same as before ✅
3. when you leave an input blank (“”), it saves that as null in the database (instead of 0) ✅

Updating other dependencies (conform-to/react etc.) caused lots of things to break (tests, types) that can be sorted out another time. 

It took literally all day, but “Years of Experience” and “Height” (in feet and inches!) are now **FULLY OPERATIONAL**  🎉

Code pushes: 
Added feet and inches inputs and logic (b14a846f2325b1731161e93dabfed64171aa57aa)
Fixed number inputs but failed tests and typechecks (3021af64e53f77b0518726f0528f3ab8f570c049)
Fixed test and typecheck errors by downgrading conform-to/react and removing deprecated conform methods (“acceptMultipleErrors: () => true”) (d06c8e09414aa7787ec54d6821d83e8a7ccd3399)
